src/main/java/com/uwaterloo/connect/service/LoginService.java

```diff
@@ -0,0 +1,93 @@
+package com.uwaterloo.connect.service;
+
+import com.uwaterloo.connect.dto.*;
+import com.uwaterloo.connect.model.Token;
+import com.uwaterloo.connect.model.User;
+import com.uwaterloo.connect.repository.UserRepository;
+import com.uwaterloo.connect.utils.EmailBuilder;
+import com.uwaterloo.connect.utils.JwtUtils;
+import lombok.AllArgsConstructor;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.RequestBody;
+
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.UUID;
+
+@Service
+@AllArgsConstructor
+public class LoginService {
+    private final AuthenticationManager authenticationManager;
+    private final JwtUtils jwtUtils;
+    private final UserService userService;
+    private final UserRepository userRepository;
+    private final TokenService tokenService;
+    private final EmailSender emailSender;
+    private final BCryptPasswordEncoder bCryptPasswordEncoder;
+
+    public LoginResponse login(@RequestBody LoginRequest request) {
+
+        Authentication authentication = authenticationManager.authenticate(new UsernamePasswordAuthenticationToken(request.getEmail(), request.getPassword()));
+        SecurityContextHolder.getContext().setAuthentication(authentication);
+        User user = (User) authentication.getPrincipal();
+        String jwtToken = jwtUtils.generateTokenFromUsername(user.getUsername());
+        return LoginResponse.builder()
+                .email(user.getEmail())
+                .jwtToken(jwtToken)
+                .code("00")
```

Comment:
@JobanThind do you think there's any benefit to maintaining separate return codes for separate requests? If not, maybe we can maintain the common stuff like the "00" code as a static string or an Enum

Reply:
Done